### DISCUSSION TITLE
Getting object length?

### DISCUSSION THREAD

**Discussion by chingc** [community/community#27125] [Actions]
<p>How do I get an object’s length?</p>
<p>I have a pull request event that shows I have two labels on a PR.</p>
<p>echo ‘${{ toJson(github.event.pull_request.labels.*.name) }}’  # shows me an array of two</p>
<p>I have tried all of these and they all return ‘false’.</p>
<p>echo ${{ github.event.pull_request.labels.*.name == 2 }}<br>
echo ${{ toJson(github.event.pull_request.labels.*.name) == 2 }}<br>
echo ${{ github.event.pull_request.labels == 2 }}<br>
echo ${{ toJson(github.event.pull_request.labels) == 2 }}</p>

**Comment by chingc** [comment]
<p>For my use case, I only needed to check whether labels existed or not so I was able to do this ugly hack:</p>
<p>jobs:</p>
<p>  test:</p>
<p>    if: “!contains(toJson(github.event.pull_request.labels), ‘[]’)”<br>
    run: # do stuff</p>
<p>But it seems like there should be an easy way to check an object’s length.  Piping it to jq would work under <code>run</code> but not in an <code>if</code>.</p>
<p>FYI: ‘github.event.pull_request.labels != null’ did not work</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p><a class="mention" href="/chingc">@chingc</a> Sorry to tell you that Github Actions context and expression syntax doesn’t provide a way to get an array or object length directly. </p>
<p>As a work around , use jq to get the labels object length in a “run” step (like you mentioned above), then you could set the value to an output variable by using <a href="https://help.github.com/en/actions/reference/development-tools-for-github-actions#set-an-output-parameter-set-output" rel="nofollow noopener">set-output</a> command. And check the output variable in step if condition. </p>
<p>There is my example:</p>
<p><img src="https://user-images.githubusercontent.com/24806478/181101955-8ccf633d-ddfa-44e0-aa13-9e180501632d.png" alt="label length.png" data-base62-sha1="3kSfmFRKyFwkjws3EuF7p9M3oTM" width="469" height="420"></p>

**Comment by chingc** [comment]
<p><a class="mention" href="/yanjingzhu">@yanjingzhu</a> Thanks for the workaround. I do hope GitHub will add a <code>length</code> function in a future update.</p>

**Comment by sjurba** [comment]
<p>Another easy trick is to check if the label at a given index is null:<br>
Check if labels is not empty:<br>
<code>if: github.event.pull_request.labels[0] != null</code><br>
Check if labels has at least 2 items:<br>
<code>if: github.event.pull_request.labels[1] != null</code></p>
<p>Etc.</p>

**Reply by dector** [reply]
Thanks! That's really less ugly than selected workaround.

**Reply by rainabba** [reply]
This really should be the selected answer. In my case, I was running against an input json param so I used `fromJson(inputs.myparam)[0] != null`

**Comment by oliora** [comment]
Such a shame there is no such function. Would make my life much easier
